File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/anmodning/AnmodningsperiodePostDto.java
package no.nav.melosys.tjenester.gui.dto.anmodning;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

import no.nav.melosys.domain.Anmodningsperiode;

public final class AnmodningsperiodePostDto {

    private List<AnmodningsperiodeSkrivDto> anmodningsperioder;

    public AnmodningsperiodePostDto(){
    }

    public AnmodningsperiodePostDto(Collection<Anmodningsperiode> anmodningsperioder) {
        this.anmodningsperioder = anmodningsperioder.stream()
            .map(AnmodningsperiodeSkrivDto::av)
            .collect(Collectors.toList());
    }

    public static AnmodningsperiodePostDto av(Collection<Anmodningsperiode> anmodningsperioder) {
        return new AnmodningsperiodePostDto(anmodningsperioder);
    }

    public List<AnmodningsperiodeSkrivDto> getAnmodningsperioder() {
        return anmodningsperioder;
    }

    public void setAnmodningsperioder(List<AnmodningsperiodeSkrivDto> anmodningsperioder) {
        this.anmodningsperioder = anmodningsperioder;
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/anmodning/AnmodningsperiodePostDto.kt
package no.nav.melosys.tjenester.gui.dto.anmodning

import no.nav.melosys.domain.Anmodningsperiode
import java.util.stream.Collectors

class AnmodningsperiodePostDto {
    var anmodningsperioder: List<AnmodningsperiodeSkrivDto>? = null

    constructor()

    constructor(anmodningsperioder: Collection<Anmodningsperiode>) {
        this.anmodningsperioder = anmodningsperioder.stream()
            .map { anmodningsperiode: Anmodningsperiode -> AnmodningsperiodeSkrivDto.av(anmodningsperiode) }
            .collect(Collectors.toList())
    }

    companion object {
        @JvmStatic
        fun av(anmodningsperioder: Collection<Anmodningsperiode>): AnmodningsperiodePostDto {
            return AnmodningsperiodePostDto(anmodningsperioder)
        }
    }
}
